RELEASE.md

```diff
@@ -2,6 +2,43 @@
 
 This document guides a contributor through creating a release of JupyterLab.
 
+## The JupyterLab Release Process
+
+JupyterLab follows [semver](https://semver.org/) for the versioning of the Python and JavaScript packages.
+
+### Release Timeline
+
+Although the commitments listed below are "best effort", the JupyterLab team tries to follow a couple of guidelines:
+
+- one major version per year, which usually includes API breaking changes
+- several minor versions per year that include new features but no API breaking changes
```

Comment:
Typical cadence would be every 2 months?

Reply:
Maybe bit too optimistic?

For reference 3.0 was released end of December 2020. And 3.3 will be out end of February 2022.

Reply:
```suggestion
- several minor versions per year that include new features but no API breaking changes, the cadence of those minor versions are typical once every 2 months (without commitment though)
```

Reply:
For the 3.x series it was more once every 4 months then?

Reply:
I have hear users reporting the 3.x series cadence was too slow, this is why defining more frequent and smaller incremental steps may help. I am fine leaving as it is, just wanted to give feedback here.

Reply:
Thanks for sharing your thoughts here.

I'm still a bit concerned about targeting "2 months", especially if beta and rc phases last 1 to 2 weeks each. Also since historically this has more been about 2 to 3 minor releases per year instead of 6.

Maybe we should aim to make more releases from now on. And if we are able to keep up with the pace adjust to time commitment.

Reply:
Sounds good. Thx.